### PR TITLE
Modify add command to accept more parameters

### PR DESCRIPTION
**Pull request by Cherlynn**
#46

### CODE REVIEW

File: .settings/org.eclipse.buildship.core.prefs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 build.commands=org.eclipse.jdt.core.javabuilder
 connection.arguments=
 connection.gradle.distribution=GRADLE_DISTRIBUTION(WRAPPER)
-connection.gradle.user.home=null
```
Comment (yamgent) [inline .settings/org.eclipse.buildship.core.prefs]: Er the removal of this line is causing issues with our Gradle build...

Could you check whether your Eclipse is unintentionally removing this line?

File: src/main/java/seedu/address/logic/parser/KeywordParser.java [hunk 2]
```diff
@@ -55,5 +55,36 @@ public KeywordParser(String... inputKeywords){
         return words;
     }
 
+    /**
+     * Similar to parse(), but for input string with one keyword
+     * Returns first match with any keyword. Associated substring is everything after the keyword
+     * @param inputString
+     * @return HashMap containing the keyword - associated substring pair
+     */
+    public HashMap<String, String> parseOne(String inputString){
```
Comment (yamgent) [inline src/main/java/seedu/address/logic/parser/KeywordParser.java]: Maybe can change to "parseForOneKeyword"

File: src/main/java/seedu/address/logic/parser/KeywordParser.java [hunk 3]
```diff
@@ -56,6 +56,44 @@ public KeywordParser(String... inputKeywords){
     }
 
     /**
+     * Like parse, but order of keywords does not matter
+     * @param string to be parsed
+     * @return HashMap containing the keyword - associated substring pairs
+     */
+    //TODO Make parsing not depend on order of input keywords
+    public HashMap<String, String> parseFree(String inputString){
```
Comment (yamgent) [inline src/main/java/seedu/address/logic/parser/KeywordParser.java]: Maybe can change to "parseKeywordsWithoutFixedOrder()"

File: .settings/org.eclipse.buildship.core.prefs (hunk 1 above)
Comment (yamgent) [inline .settings/org.eclipse.buildship.core.prefs:4]: Er the removal of this line is causing issues with our Gradle build...

Could you check whether your Eclipse is unintentionally removing this line?

**Review by yamgent — CHANGES_REQUESTED** [review]
Needs changes as pointed out in comments.

File: .settings/org.eclipse.buildship.core.prefs (hunk 1 above)
Comment (Cherlynn) [reply .settings/org.eclipse.buildship.core.prefs:4]: restored the missing line

File: src/main/java/seedu/address/logic/parser/KeywordParser.java (hunk 2 above)
Comment (Cherlynn) [reply src/main/java/seedu/address/logic/parser/KeywordParser.java]: Renamed. Though I had planned to replace the three with one parse method later on, when I'm more certain it works properly for all these cases.

**Review by yamgent — APPROVED** [review]

**Pull request merged**